src/Polynomial.java

```diff
@@ -0,0 +1,674 @@
+    
+    /** return a new Polynomial which is the derivative of its original value.
+
+	@return derivative of this Polynomial
+
+    */
+
+    public Polynomial derivative () {
+	if(this.size()==1) return new Polynomial(new int[]{0});
+	else{
+	    int [] newArr = new int[this.size()-1];
+	    for(int i = 1;i<this.size();i++){
+		newArr[i-1]=this.get(i)*i;
+	    }
+	    Polynomial newP = new Polynomial (Polynomial.highToLow(newArr));
+	    return newP;
+	}	
+    }
+    
+    /** return a new Polynomial which has as its value the 
+	this polynomial plus the one passed in as a parameter.
+
+	@param p the Polynomial to add to this one
+	@return sum of this Polynomial and p
+
+    */
+
+    public Polynomial plus (Polynomial p) {
+	int newSize = Math.max(this.size(),p.size());
+	int smallSize = Math.min(this.size(),p.size());
+	int [] newArr = new int[newSize];
+	for(int i=0; i<smallSize; i++){
+	    newArr[i] = this.get(i)+p.get(i);
+	}
+	if(this.size() > smallSize){
+	    for(int i = smallSize; i<newSize; i++){
+		newArr[i] = this.get(i);
+	    }
+	}else if(p.size() > smallSize){
+	    for(int i = smallSize; i<newSize; i++){
+		newArr[i] = p.get(i);
+	    }
+	}
+	
+	Polynomial newP = new Polynomial (Polynomial.highToLow(newArr));
+	return newP;
+    }
+    
+    
+    /** return a new Polynomial which has as its value the 
+	this polynomial times the one passed in as a parameter.
+
+	@param p the Polynomial to multiply this one by
+	@return product of this Polynomial and p
+
+    */
+
+    public Polynomial times (Polynomial p) {
+	Polynomial newP = new Polynomial (new int[]{0});
+	int [] newA;
+	Polynomial oneP;
+	for(int i = 0; i<this.size();i++){
+	    newA = new int [p.size()+i];
+	    for(int k = 0; k < i; k++){
+		newA[k] = 0;
+	    }
+	    for(int j = 0; j<p.size();j++){
+		newA[j+i]=this.get(i)*p.get(j);
+	    }
+	    oneP = new Polynomial(Polynomial.highToLow(newA));
+	    newP = newP.plus(oneP);
+	}
+	return newP; 
+
+    }
+
+    /** return a new Polynomial which has as its value the 
+	this polynomial minus the one passed in as a parameter.
+
+	@param p the Polynomial to subtract from this one
+	@return the result of this Polynomial minus p
+
+    */
+
+
+    public Polynomial minus (Polynomial p) {
+	Polynomial negOne = new Polynomial (new int[]{-1});
+	return this.plus(p.times(negOne));
+	
+    }
+
+    /** Print Usage message for Polynomial main 
+     */
+
+    public static void usage() {
+	System.err.println("Usage: java Polynomial 'string' ");
+	System.err.println(" java Polynomial 'string' ");
+    }
+
+    /** 
+	Main for testing constructing Polynomials from strings,
+	and for testing plus, minus and times.
+
+	At Unix command line, use single quotes to make the entire
+	Polynomial be a single argument, and use 
+	\* when operating is * (to avoid
+	having * expanded as a wildcard.)  For example:
+	<code>java -cp build Polynomial 'x^2 + 2x + 3' \* 'x - 4'</code>
+     */
+
+    public static void main (String [] args) {
```

Comment:
This class should not contain a main function as PolynomialDerivative contains the main